terraform/gt-site.tf

```diff
@@ -0,0 +1,91 @@
+provider "aws" {
+  profile = "${var.profile}" # Assumes you have valid AWS credentials.
+  region  = "${var.region}"
+}
+
+# --- ECS RESOURCES --- #
+
+resource "aws_ecs_cluster" "gt-site-cluster" {
+  name = "gt-site-cluster"
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+# `containers.json` is analogous to `docker-compose.yml`.
+# All the container settings are there.
+resource "aws_ecs_task_definition" "gt-site-cluster-task" {
+  family                = "gt-site-cluster"
+  container_definitions = "${file("containers.json")}"
+}
+
+resource "aws_ecs_service" "gt-site-cluster-service" {
+  name                               = "gt-site-cluster"
+  cluster                            = "${aws_ecs_cluster.gt-site-cluster.id}"
+  task_definition                    = "${aws_ecs_task_definition.gt-site-cluster-task.family}:${aws_ecs_task_definition.gt-site-cluster-task.revision}"
+  desired_count                      = "${var.desired_instance_count}"
+  iam_role                           = "${var.ecs_service_role}"
+  deployment_minimum_healthy_percent = "0"
+  deployment_maximum_percent         = "100"
+
+  load_balancer {
+    elb_name       = "${aws_elb.gt-site-elb.name}"
+    container_name = "gtsite-nginx"
+    container_port = 8080
+  }
+}
+
+# A load balancer is necessary to launch the EC2 manager instance properly.
+resource "aws_elb" "gt-site-elb" {
```

Comment:
We should be using an `aws_alb` resources vs. the legacy `aws_elb`. The set of resources necessary to make that happen look something like:

```hcl
resource "aws_alb" "app_server" {
  security_groups = ["${aws_security_group.app_server_alb.id}"]
  subnets         = ["${module.vpc.public_subnet_ids}"]
  name            = "alb${var.environment}AppServer"

  tags {
    Name        = "albAppServer"
    Project     = "${var.project}"
    Environment = "${var.environment}"
  }
}

resource "aws_alb_target_group" "app_server_https" {
  # Name can only be 32 characters long, so we MD5 hash the name and
  # truncate it to fit.
  name = "tf-tg-${replace("${md5("${var.environment}HTTPSAppServer")}", "/(.{0,26})(.*)/", "$1")}"

  health_check {
    healthy_threshold   = "3"
    interval            = "30"
    protocol            = "HTTP"
    timeout             = "3"
    path                = "/healthcheck/"
    unhealthy_threshold = "2"
  }

  port     = "443"
  protocol = "HTTP"
  vpc_id   = "${module.vpc.id}"

  tags {
    Name        = "tg${var.environment}HTTPSAppServer"
    Project     = "${var.project}"
    Environment = "${var.environment}"
  }
}

resource "aws_alb_listener" "app_server_https" {
  load_balancer_arn = "${aws_alb.app_server.id}"
  port              = "443"
  protocol          = "HTTPS"
  certificate_arn   = "${var.ssl_certificate_arn}"

  default_action {
    target_group_arn = "${aws_alb_target_group.app_server_https.id}"
    type             = "forward"
  }
}
```